### PR TITLE
Use explicit keyword argument in Intersection.__new__

### LINKED ISSUE #14030: Functions should list explicit keyword arguments when possible

### PR DESCRIPTION
**Pull request by Jay-Patankar**
See #14030 

Removed the **kwargs from the function signature of new(), replacing it with an equivalent default argument.

#### Release Notes

* sets
  * `Intersection.__new__` now accepts only an explicit keyword-only argument `evaluate` rather than `**kwargs`.

### CODE REVIEW

**Comment by Jay-Patankar** [comment]
@oscarbenjamin  @asmeurer @1e9abhi1e10 This PR is passing all the tests except for the authors file error. I added my name and email to it .....still not working . It asks for missing names of some other contributors....so I added their names also to .mailmap file; still no avail. What should be done in such cases? I am just one step away from my first PR !!

File: .mailmap [hunk 1]
```diff
@@ -1537,3 +1537,8 @@ zzj <29055749+zjzh@users.noreply.github.com>
 Øyvind Jensen <jensen.oyvind@gmail.com>
 Łukasz Pankowski <lukpank@o2.pl>
 彭于斌 <1931127624@qq.com>
+Jay Patankar <patankarjays@gmail.com>
```
Comment (1e9abhi1e10) [inline .mailmap:1540]: you need to only add your name and email in the `.mailmap` file and then run the following command `python bin/mailmap_check.py` or read [here](https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file)

**Comment by oscarbenjamin** [comment]
I think the problem with the AUTHORS check is somehow to do with the fact that I just updated the AUTHORS file. Let me fix it for you.

File: .mailmap (hunk 1 above)
Comment (Jay-Patankar) [reply .mailmap]: @1e9abhi1e10  I added my name and ran the python bin/mailmap_check.py script . Still I am getting following error:
`Run bin/mailmap_check.py
This author is in the AUTHORS file but not .mailmap:
Charles Harris <erdos4d@gmail.com>
This author is in the AUTHORS file but not .mailmap:
Tejaswini Sanapathi <sastejaswini2002@gmail.com>
This author is in the AUTHORS file but not .mailmap:
Devansh <be19b002@smail.iitm.ac.in>
This author is in the AUTHORS file but not .mailmap:
Aaron Gokaslan <aaronGokaslan@gmail.com>

For instructions on updating the .mailmap file see:
https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file

The following authors will be added to the AUTHORS file at the time of
the next SymPy release.

	Jay Patankar <patankarjays@gmail.com>
Error: Process completed with exit code 1.`

What should be done? Also is there any way of running a particular test once I commit my changes as it is unnecessarily taking a lot of time everytime to execute all the tests  ....Can I rerun only a specific test keeping the pass status of all previous tests intact?

**Comment by Jay-Patankar** [comment]
> I think the problem with the AUTHORS check is somehow to do with the fact that I just updated the AUTHORS file. Let me fix it for you.
Okk

**Comment by oscarbenjamin** [comment]
I've fixed it by rebasing the PR onto master. I had to do a force push so you will need to update your branch locally. I suggest deleting your local branch and then checking it again from the remote:
```
git checkout master
git branch -d first-contri
git fetch origin
git checkout first-contri
```

**Comment by Jay-Patankar** [comment]
> 

Is it required to delete this branch and again create it .....because the authors file which was giving error previously has passed the tests after you made changes....and rest of the tests will also be passed soon . So is it okay if I wait for all tests to pass?

**Comment by oscarbenjamin** [comment]
> Is it required to delete this branch and again create it

I just mean to delete the branch locally. It will still exist in github so after deleting it you can use fetch and checkout to recreate the local branch.

**Comment by Jay-Patankar** [comment]
My first PR has finally passed all tests.......requesting @oscarbenjamin  @1e9abhi1e10  @tjl @asmeurer to merge

**Comment by Jay-Patankar** [comment]
@smichr 
@oscarbenjamin 
@asmeurer 
@mattpap 
@mrocklin 
@sylee957 
@bjodah 
@Upabjojr 
@jrioux 
Can someone review this PR ? It has passed the tests but not got merged? Can someone confirm whether I can proceed for refactoring functions in other modules as expected by issue #14030 , as done in this PR.

**Comment by bjodah** [comment]
@Jay-Patankar thank you for working on this. It is much appreciated. However, I kindly ask you to refrain from pinging contributors en masse, this is an open-source project and many people are busy with their everyday jobs. There's a gitter channel and a mailing list which might be a better place to find more interactive feedback.

**Comment by Jay-Patankar** [comment]
> @Jay-Patankar thank you for working on this. It is much appreciated. However, I kindly ask you to refrain from pinging contributors en masse, this is an open-source project and many people are busy with their everyday jobs. There's a gitter channel and a mailing list which might be a better place to find more interactive feedback.

Okk will take care of that @bjodah . Just wanted to have a confirmation whether my contribution is as expected by the issue #14030 . If yes I can proceed with solving other instances of this issue in the codebase.

File: sympy/sets/sets.py [hunk 2]
```diff
@@ -1474,8 +1474,9 @@ def identity(self):
     def zero(self):
         return S.EmptySet
 
-    def __new__(cls, *args, **kwargs):
-        evaluate = kwargs.get('evaluate', global_parameters.evaluate)
+    def __new__(cls, *args , evaluate=None):
+        if evaluate==None:
+            evaluate=global_parameters.evaluate
```
Comment (sylee957) [inline sympy/sets/sets.py:1479]:
```suggestion
        if evaluate == None:
            evaluate = global_parameters.evaluate
```

**Comment by Jay-Patankar** [comment]
@sylee957 I had already added my name and email to .mailmap. Now again the authors file check is failing ....it is asking me to add my name again with  a strange email as : <107458263+Jay-Patankar@users.noreply.github.com>. Do  authors need to add these details before every commit? Or can you elaborate why this is happening?

**Comment by oscarbenjamin** [comment]
https://github.com/sympy/sympy/blob/75691e5cbdb3dfda4c8164a217bac6a6434404f6/.mailmap#L113-L118

**Comment by Jay-Patankar** [comment]
> https://github.com/sympy/sympy/blob/75691e5cbdb3dfda4c8164a217bac6a6434404f6/.mailmap#L113-L118

Thanks a lot

**Comment by Jay-Patankar** [comment]
@sylee957  @bjodah  Can someone say why one of the tests is not passing.....I mean I tried to see the log but its just mentioned that exit code is 1. Also is there any way of running a particular test once I commit my changes as it is unnecessarily taking a lot of time everytime to execute all the tests ....Can I rerun only a specific test keeping the pass status of all previous tests intact?

**Comment by bjodah** [comment]
It looks like it's the 3.12 alpha test that is failing. I wouldn't worry about it since it is not a required test.

**Comment by Jay-Patankar** [comment]
> It looks like it's the 3.12 alpha test that is failing. I wouldn't worry about it since it is not a required test.

@bjodah  ok thanks ! Is there any fixed number of reviewers  that must review my PR in order for it to get merged ....like whats the next step that I should take in order to have this PR merged?

**Comment by oscarbenjamin** [comment]
I edited the OP so that it doesn't say that it "fixes" the issue because the issue applies widely across the codebase and this PR only changes one function.

**Comment by oscarbenjamin** [comment]
I also changed the title and edited the release note. The purpose of the PR title and the release note is so that someone looking at this can understand what has been changed. Previously neither the title nor the release note even mentioned the fact that this is a change to the `Intersection` class.

**Comment by Jay-Patankar** [comment]
> I edited the OP so that it doesn't say that it "fixes" the issue because the issue applies widely across the codebase and this PR only changes one function.

Ok thanks . May I know when regarding the process of merging my PR ?

**Comment by Jay-Patankar** [comment]
@oscarbenjamin  Thanks for merging It motivates a lot!!

**Pull request merged**